p2p/crypto/ed25519.go

```diff
@@ -0,0 +1,102 @@
+package crypto
+
+import (
+	"errors"
+	"fmt"
+	"github.com/agl/ed25519"
+	"io"
+
+	proto "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/goprotobuf/proto"
+
+	pb "github.com/jbenet/go-ipfs/p2p/crypto/internal/pb"
+)
+
+type Ed25519PrivateKey struct {
+	sk *[ed25519.PrivateKeySize]byte
+	pk *[ed25519.PublicKeySize]byte
+}
+
+type Ed25519PublicKey struct {
+	k *[ed25519.PublicKeySize]byte
+}
```

Comment:
I was thinking of implementing it the same way for RSA and ed25519. Do you think it is also desirable to put the RSA functions in a separate package as well ?

Also, I'm a bit reluctant here because we would have two packages containing different things and having the same name. the ed25519 from github.com/agl/ed25519 an the version from p2p/crypto/ed25519. Do you think this should be avoided, or not ? Same with the rsa package.